samples/fpgaperf_counter/fpgaperf_counter.h

```diff
@@ -61,6 +70,28 @@ typedef struct {
 	perf_events_type *perf_events;
 } fpga_perf_counter;
 
+/**
+ * Initialize the pthread mutex.
+ *
+ * @param[inout] fpga_perf  Returns the fpga_perf_counter struct
+ *
+ * @returns FPGA_OK on success. FPGA_INVALID_PARAM if any of the supplied
+ * parameters is invalid. FPGA_EXCEPTION if an internal exception occurred
+ * while trying to access the handle.
+ */
+fpga_result fpgaPerfCreateHandle(fpga_perf_counter *fpga_perf);
+
+/**
+ * Destroy the pthread mutex.
```

Comment:
I think this is an implementation detail that doesn't really summarize what this function does.
IOW, it's too specific .

Reply:
Also, I'm a little confused why we have two API functions that initialize part of the data structure.